core/src/com/cloud/agent/resource/virtualnetwork/facade/AbstractConfigItemFacade.java

```diff
@@ -104,13 +109,25 @@ public static AbstractConfigItemFacade getInstance(final Class<? extends Network
         return instance;
     }
 
+
+    private static String appendUuidToJsonFiles(final String filename) {
+        String remoteFileName = new String(filename);
```

Comment:
Why are is a ``new String(filename)`` being done?  This avoids the constant pool, and is extremely expensive.  Consider the following usage of ``StringBuilder`` to avoid unnecessary pressure on the ``String`` constant pool due to the ``new`` call and string concatenation:

```
final StringBuilder remoteFileName = new StringBuilder(filename);
if (filename.endWith("json")) {
    remoteFileName.append(".");
    remoteFileName.append(UUID.randomUUID()); // implicit call to toString
}
return remoteFileName.toString();
```